### ISSUE TITLE
Support Bluetooth Scan / Scan Results

### ISSUE THREAD

**Issue by stboch** [leikoilja/ha-google-home#123]
Back at the start of the GHLocalApi several of us where starting to build a triangulation based presence system based on bluetooth results. Would love to help get this working again as we now have access again. 

[Get Scan Results](https://rithvikvibhu.github.io/GHLocalApi/#operation/GetScanResults)
[Initiate Scan](https://rithvikvibhu.github.io/GHLocalApi/#operation/Scanfordevices)

This would likely require creating new entities as each rescan interval potentially identifies new devices, it all depends.

**Comment by KapJI** [comment]
I remember old official `google_home` integration had this feature. Can we just restore that deleted code?

**Comment by stboch** [comment]
Possibly here is the historical branch for reference 
https://github.com/home-assistant/core/tree/a2589f56e1ac87023123dfb98dffc968073c6ca5/homeassistant/components/googlehome

**Comment by stboch** [comment]
It also looks like someone updated https://github.com/ludeeus/googledevices to support tokens in the client calls.

**Comment by stboch** [comment]
Not sure if @Drakulix has seen this component or if you have seen https://github.com/Drakulix/googlehome might be some opporutinity to increase the development effort and combine the projects :)

**Comment by leikoilja** [comment]
Love the idea of introducing a triangulation presence system by use of Bluetooth based devices as you propose, we could definitely do it if someone is interested in taking a lead on that :) 

Oh nice, haven't seen @Drakulix fork. It doesn't seem to be a very active project yet, but it would be definitely worth giving a shot to try merging the project and inviting @Drakulix to combine efforts with us 🚀

**Comment by Leatherface75** [comment]
Any news about this?

**Comment by leikoilja** [comment]
Not really, @Leatherface75. We don't have anyone interested/available in picking it up yet. @Drakulix agreed to help and mentor anyone who would like to take a lead on this :) So if you have skills and interest, we would be happy you join us ;)

**Comment by Leatherface75** [comment]
Sorry i don't have the skills for bigger things just for small changes in already existing code.

**Comment by Leatherface75** [comment]
Something new about this problem yet?

**Comment by ArnyminerZ** [comment]
Well, maybe @Drakulix can help sticking the code from [Drakulix/googlehome](https://github.com/Drakulix/googlehome) to this repo, so we have the same features.

Also the guys at [ludeeus](https://github.com/ludeeus/googledevices) can help on creating a common package for all the API calls as discussed in the Developers Discussion panel some weeks ago. Instead of creating a new package from zero, maybe we can talk to them, and create a common codebase. Anyway, this should be discussed somewhere else.

**Comment by yevon** [comment]
This would be awesome to have as I have bluetooth speakers on every room, I might help with this but I have no experience developing for home assistant.

**Comment by leikoilja** [comment]
@yevon, that sounds amazing! We have a cool team around that can certainly help you around. Just skim through how the current integration is implemented and ask any questions in [discussions section](https://github.com/leikoilja/ha-google-home/discussions) if you might have. Also whenever you build something, try splitting those in smaller PR chunks so it would be easier for us to review and help around if you get stuck with anything :)

**Comment by Leatherface75** [comment]
Anything new?

**Comment by Leatherface75** [comment]
Anything new?

**Comment by ChrisAllen1982** [comment]
I have begun implementing this feature on a fork. When it is in a position to be merged, I'll submit a pull request. It's at https://github.com/ChrisAllen1982/ha-google-home.

**Comment by Leatherface75** [comment]
Seems to work but it doesn't find all bluetooth devices.

**Comment by stboch** [comment]
@ChrisAllen1982 hey turn on issue tracker in your repo so we can discuss things there, I have some input and thoughts to discuss rather not pollute this thread.

**Comment by DurgNomis-drol** [comment]
Please keep all discussion relating to this integration here, you can open a discussion instead if you need it 😊 This will make it easier in the future for everyone to see what you guys where thinking and we may all learn something 😊

**Comment by leikoilja** [comment]
@DurgNomis-drol has a very valid point :) Feel free to keep the conversation here or open a new discussion about Bluetooth scanning under discussion panel: https://github.com/leikoilja/ha-google-home/discussions

@ChrisAllen1982, it's awesome to hear that you have started tweaking things into the direction of Bluetooth scanning 💥  Feel free to open smaller chunks of progress PRs for earlier reviews or/and any help if you get stuck with anything :)

**Comment by ChrisAllen1982** [comment]
Ok, I shall do. I've currently got reporting of visible Bluetooth devices working (under the device), so I'll add a PR for that once I've updated the docs. Then I'll go on to adding the device tracking functionality.

**Comment by TheFes** [comment]
@ChrisAllen1982 any news on this? It would be a great Christmas gift :)

**Comment by Leatherface75** [comment]
Yeah i wonder that too

**Comment by mshuflin** [comment]
Hi, @ChrisAllen1982, any chance you will be able to add some of the functionality?

**Comment by Leatherface75** [comment]
Still a few months anything new?

**Comment by yevon** [comment]
Could this be integrated with just released home assistant bluetooth proxy for scanning tags?

**Comment by HarvsG** [comment]
https://github.com/leikoilja/ha-google-home/pull/590
